src/tags/master_playlist/media.rs

```diff
@@ -238,9 +238,12 @@ impl ExtXMediaBuilder {
             ).to_string());
         }
 
-        if self.is_default.unwrap_or(false) && !self.is_autoselect.unwrap_or(false) {
+        if self.is_default.unwrap_or(false)
+            && self.is_autoselect.is_some()
+            && !self.is_autoselect.unwrap()
```

Comment:
```suggestion
        if self.is_default.unwrap_or(false) && self.is_autoselect.map_or(false, |b| !b)
```

I think this is a bit cleaner :)